### PR TITLE
chore(docs): remove outdated content

### PR DESCRIPTION
**Pull request by jhchabran**
As per https://github.com/sourcegraph/deploy-sourcegraph-cloud/pull/18520 we now serve the docsite content from the `v5.3.9104` branch, we can now remove the remaining content, with the exception of the automatically generated docs, which are still present in the docsite structure as well as the dev docs. 

I have updated the sidebar to explain what is going on: 

CleanShot 2024-04-08 at 15 54 31@2x

## Test plan

See https://github.com/sourcegraph/sourcegraph/pull/61684 and https://github.com/sourcegraph/deploy-sourcegraph-cloud/pull/18520 where I have purposely removed a bit of the content, to ensure the docsite would properly serve from the specified branch instead of `main`.

### CODE REVIEW

**Review by burmudar — APPROVED** [review]

**Comment by jhchabran** [comment]
Dear reviewers, best to go commit by commit :)

**Review by eseliger — APPROVED** [review]
chrome crashes trying to review this, so I have to trust you :p

File: doc/sidebar.md [hunk 1]
```diff
@@ -25,127 +25,21 @@ Keep it as a single list with at most 2 levels. (Anything else may not render co
...
-
-<h3 class="heading">Integrations</h3>
-
-- [Overview](integration/index.md)
-- [Browser extension](integration/browser_extension/index.md)
-- [Editors](integration/editor.md)
-- [Browser search engine](integration/browser_extension/how-tos/browser_search_engine.md)
+<p>⚠️ <b>You are browsing a trimmed down version of the docsite</b> which is available for the convenience of browsing automatically generated documentation.</p>
+<p>On <a href="https://docs.sourcegraph.com">docs.sourcegraph.com</a> the content is being served from the <a href="">v5.3.9104</a> branch which
+is the last point in time at which we had this content avaible, in order to maintain contuinity until the legacy docsite if fully removed.</p>
```
[the remark this answers is not in the document]
Comment (burmudar) [reply doc/sidebar.md:30]: also `contuinity`->`continuity`

[the remark this answers is not in the document]
Comment (burmudar) [reply doc/sidebar.md:30]:
```suggestion
is the last point in time at which we had this content avaible, in order to maintain contuinity until the legacy docsite is fully removed.</p>
```

**Pull request merged**